libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/LayoutSyncDownTarget.java

```diff
@@ -98,7 +98,6 @@ public JSONArray startFetch(SyncManager syncManager, long maxTimeStamp) throws I
         final RestResponse response = syncManager.sendSyncWithSmartSyncUserAgent(request);
         final JSONObject responseJSON = response.asJSONObject();
         if (responseJSON != null) {
-            responseJSON.put(SOBJECT_TYPE, objectType);
```

Comment:
Should we keep that, so that you don't have to parse that value out of the id field?

Reply:
We don't parse it out of the `Id` field. It's available in the callback since `objectType` is an instance variable.

Reply:
See [here](https://github.com/bhariharan/SalesforceMobileSDK-Android/blob/dev/libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/LayoutSyncManager.java#L182) - we have `objectType` and we pass it back in the callback.

Reply:
I was thinking - if one was to read directly from the db - maybe that’s not a valid use case.